include/deal.II/base/tensor.h

```diff
@@ -2112,6 +2112,48 @@ inline DEAL_II_ALWAYS_INLINE Tensor<1, dim, Number>
 }
 
 
+/**
+ * Return the cross product of 2 vectors in 3d. Same as above for the case in
+ * which one of the vectors is complex and not other.
+ */
+template <int dim, typename Number>
+inline DEAL_II_ALWAYS_INLINE Tensor<1, dim, std::complex<Number>>
```

Comment:
Can't you just allow different `Number` types for the two arguments via a second template parameter?
The return value type would then be
```
Tensor<1, dim, typename ProductType<Number1, Number2>::type>
```

Reply:
This is a much better solution :)
I updated the PR